src/aerie_cli/aerie_client.py

```diff
@@ -1654,8 +1654,28 @@ def __expand_activity_arguments(self, plan: ActivityPlanRead, full_args: str = N
                 activity.arguments = ApiEffectiveActivityArguments.from_dict(
                     resp).arguments
         return plan
+    
+    def add_constraint_tag(self, constraint_id: int, tag_name: str):
+        add_constraint_tag_query = """
+        mutation AddTagToConstraint($constraint_id: Int, $tag_id: Int) {
+            insert_constraint_tags(objects: {constraint_id: $constraint_id, tag_id: $tag_id}) {
+                returning {
+                    tag_id
+                }
+            }
+        }
+        """
+        
+        #add tag to constraint
+        resp = self.aerie_host.post_to_graphql(
+            add_constraint_tag_query, 
+            constraint_id=constraint_id, 
+            tag_id=self.get_tag_id_by_name(tag_name)
+        )
 
-    def upload_constraint(self, constraint):
+        return resp['returning'][0]
+
+    def upload_constraint(self, constraint, tags):
```

Comment:
Can `tags` either be an optional argument or something the user is required to manually add separately?

Reply:
I can make it a default parameter. There is the `add_constraint_tag` method that allows user to manually add it separately as well.